src/x509/clu_request_setup.c

```diff
@@ -116,8 +116,12 @@ static int _wolfSSL_X509_print_serial(WOLFSSL_BIO* bio, WOLFSSL_X509* x509,
                 }
             }
         }
+        
+        return WOLFSSL_SUCCESS;
     }
-    return WOLFSSL_SUCCESS;
+    else {
+        return WOLFSSL_FAILURE; 
```

Comment:
It's not necessarily a fail case if a serial number is not set. Would leave the sanity check of the certificate 'correctness' up to certificate parsing code rather than the print out functions.

Reply:
Restored.